src/Bridges/Constraint/semi_to_binary.jl

```diff
@@ -0,0 +1,191 @@
+            (MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}),
+            (MOI.SingleVariable, MOI.Integer),
+            ]
+end
+
+function concrete_bridge_type(::Type{<:SemiToBinaryBridge{T}},
+                              ::Type{MOI.SingleVariable},
+                              ::Type{S}) where {T, S<:SemiSets}
+    return SemiToBinaryBridge{T, S}
+end
+
+# Attributes, Bridge acting as a model
+
+function MOI.get(::SemiToBinaryBridge, ::MOI.NumberOfVariables)
+    return 1
+end
+
+function MOI.get(b::SemiToBinaryBridge, ::MOI.ListOfVariableIndices)
+    return [b.binary_variable_index]
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.SingleVariable, MOI.ZeroOne}) where {T, S}
+    return 1
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.SingleVariable, MOI.Integer}) where {T, S<:MOI.Semiinteger}
+    return 1
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}}) where {T, S}
+    return 1
+end
+
+function MOI.get(::SemiToBinaryBridge{T, S},
+    ::MOI.NumberOfConstraints{MOI.ScalarAffineFunction{T}, MOI.LessThan{T}}) where {T, S}
+    return 1
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.SingleVariable, MOI.ZeroOne}) where {T, S}
+    return [b.binary_constraint_index]
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.SingleVariable, MOI.Integer}) where {T, S<:MOI.Semiinteger}
+    return [b.integer_index]
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T}, MOI.LessThan{T}}) where {T, S}
+    return [b.upper_bound_index]
+end
+
+function MOI.get(b::SemiToBinaryBridge{T, S},
+    ::MOI.ListOfConstraintIndices{MOI.ScalarAffineFunction{T}, MOI.GreaterThan{T}}) where {T, S}
+    return [b.lower_bound_index]
+end
+
+function MOI.supports(
+    ::MOI.ModelLike,
+    ::MOI.ConstraintPrimalStart,
+    ::Type{<:SemiToBinaryBridge})
+    return true
+end
+
+function MOI.get(model::MOI.ModelLike, attr::MOI.ConstraintPrimalStart, bridge::SemiToBinaryBridge)
+    return MOI.get(model, MOI.VariablePrimalStart(), bridge.variable_index)
+end
+
+function MOI.set(model::MOI.ModelLike, attr::MOI.ConstraintPrimalStart,
+                 bridge::SemiToBinaryBridge{T}, value) where {T}
+    MOI.set(model, MOI.VariablePrimalStart(), bridge.variable_index, value)
+    bin_value = ifelse(value == 0.0, 0.0, 1.0)
+    MOI.set(model, MOI.VariablePrimalStart(), bridge.binary_variable_index, bin_value)
+    MOI.set(model, MOI.ConstraintPrimalStart(),
+        bridge.upper_bound_index, value - bridge.semi_set[].upper * bin_value)
+    MOI.set(model, MOI.ConstraintPrimalStart(),
+        bridge.lower_bound_index, value - bridge.semi_set[].lower * bin_value)
+    return
+end
+
+function MOI.set(model::MOI.ModelLike, attr::MOI.ConstraintSet,
```

Comment:
Missing getters for `ConstraintFunction` and `ConstraintSet`

Reply:
They are just out of order, I will reorder the functions.a